// @racehub-io/f1-telemetry-client/build/src/mocks/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PacketCarSetupData_1 = require("./2018/PacketCarSetupData");
exports.PACKET_CAR_SETUP_DATA_BUFFER_2018 = PacketCarSetupData_1.PACKET_CAR_SETUP_DATA_BUFFER_2018;
exports.PACKET_CAR_SETUP_DATA_PARSED_2018 = PacketCarSetupData_1.PACKET_CAR_SETUP_DATA_PARSED_2018;
const PacketCarStatusData_1 = require("./2018/PacketCarStatusData");
exports.PACKET_CAR_STATUS_DATA_BUFFER_2018 = PacketCarStatusData_1.PACKET_CAR_STATUS_DATA_BUFFER_2018;
exports.PACKET_CAR_STATUS_DATA_PARSED_2018 = PacketCarStatusData_1.PACKET_CAR_STATUS_DATA_PARSED_2018;
const PacketCarTelemetryData_1 = require("./2018/PacketCarTelemetryData");
exports.PACKET_CAR_TELEMETRY_DATA_BUFFER_2018 = PacketCarTelemetryData_1.PACKET_CAR_TELEMETRY_DATA_BUFFER_2018;
exports.PACKET_CAR_TELEMETRY_DATA_PARSED_2018 = PacketCarTelemetryData_1.PACKET_CAR_TELEMETRY_DATA_PARSED_2018;
const PacketEventData_1 = require("./2018/PacketEventData");
exports.PACKET_EVENT_DATA_BUFFER_2018 = PacketEventData_1.PACKET_EVENT_DATA_BUFFER_2018;
exports.PACKET_EVENT_DATA_PARSED_2018 = PacketEventData_1.PACKET_EVENT_DATA_PARSED_2018;
const PacketHeader_1 = require("./2018/PacketHeader");
exports.PACKET_HEADER_BUFFER_2018 = PacketHeader_1.PACKET_HEADER_BUFFER_2018;
exports.PACKET_HEADER_PARSED_2018 = PacketHeader_1.PACKET_HEADER_PARSED_2018;
const PacketLapData_1 = require("./2018/PacketLapData");
exports.PACKET_LAP_DATA_BUFFER_2018 = PacketLapData_1.PACKET_LAP_DATA_BUFFER_2018;
exports.PACKET_LAP_DATA_PARSED_2018 = PacketLapData_1.PACKET_LAP_DATA_PARSED_2018;
const PacketMotionData_1 = require("./2018/PacketMotionData");
exports.PACKET_MOTION_DATA_BUFFER_2018 = PacketMotionData_1.PACKET_MOTION_DATA_BUFFER_2018;
exports.PACKET_MOTION_DATA_PARSED_2018 = PacketMotionData_1.PACKET_MOTION_DATA_PARSED_2018;
const PacketParticipantsData_1 = require("./2018/PacketParticipantsData");
exports.PACKET_PARTICIPANTS_DATA_BUFFER_2018 = PacketParticipantsData_1.PACKET_PARTICIPANTS_DATA_BUFFER_2018;
exports.PACKET_PARTICIPANTS_DATA_PARSED_2018 = PacketParticipantsData_1.PACKET_PARTICIPANTS_DATA_PARSED_2018;
const PacketSessionData_1 = require("./2018/PacketSessionData");
exports.PACKET_SESSION_DATA_BUFFER_2018 = PacketSessionData_1.PACKET_SESSION_DATA_BUFFER_2018;
exports.PACKET_SESSION_DATA_PARSED_2018 = PacketSessionData_1.PACKET_SESSION_DATA_PARSED_2018;
const PacketCarSetupData_2 = require("./2019/PacketCarSetupData");
exports.PACKET_CAR_SETUP_DATA_BUFFER_2019 = PacketCarSetupData_2.PACKET_CAR_SETUP_DATA_BUFFER_2019;
exports.PACKET_CAR_SETUP_DATA_PARSED_2019 = PacketCarSetupData_2.PACKET_CAR_SETUP_DATA_PARSED_2019;
const PacketCarStatusData_2 = require("./2019/PacketCarStatusData");
exports.PACKET_CAR_STATUS_DATA_BUFFER_2019 = PacketCarStatusData_2.PACKET_CAR_STATUS_DATA_BUFFER_2019;
exports.PACKET_CAR_STATUS_DATA_PARSED_2019 = PacketCarStatusData_2.PACKET_CAR_STATUS_DATA_PARSED_2019;
const PacketCarTelemetryData_2 = require("./2019/PacketCarTelemetryData");
exports.PACKET_CAR_TELEMETRY_DATA_BUFFER_2019 = PacketCarTelemetryData_2.PACKET_CAR_TELEMETRY_DATA_BUFFER_2019;
exports.PACKET_CAR_TELEMETRY_DATA_PARSED_2019 = PacketCarTelemetryData_2.PACKET_CAR_TELEMETRY_DATA_PARSED_2019;
const PacketEventData_2 = require("./2019/PacketEventData");
exports.PACKET_EVENT_DATA_BUFFER_2019 = PacketEventData_2.PACKET_EVENT_DATA_BUFFER_2019;
exports.PACKET_EVENT_DATA_PARSED_2019 = PacketEventData_2.PACKET_EVENT_DATA_PARSED_2019;
const PacketHeader_2 = require("./2019/PacketHeader");
exports.PACKET_HEADER_BUFFER_2019 = PacketHeader_2.PACKET_HEADER_BUFFER_2019;
exports.PACKET_HEADER_PARSED_2019 = PacketHeader_2.PACKET_HEADER_PARSED_2019;
const PacketLapData_2 = require("./2019/PacketLapData");
exports.PACKET_LAP_DATA_BUFFER_2019 = PacketLapData_2.PACKET_LAP_DATA_BUFFER_2019;
exports.PACKET_LAP_DATA_PARSED_2019 = PacketLapData_2.PACKET_LAP_DATA_PARSED_2019;
const PacketMotionData_2 = require("./2019/PacketMotionData");
exports.PACKET_MOTION_DATA_BUFFER_2019 = PacketMotionData_2.PACKET_MOTION_DATA_BUFFER_2019;
exports.PACKET_MOTION_DATA_PARSED_2019 = PacketMotionData_2.PACKET_MOTION_DATA_PARSED_2019;
const PacketParticipantsData_2 = require("./2019/PacketParticipantsData");
exports.PACKET_PARTICIPANTS_DATA_BUFFER_2019 = PacketParticipantsData_2.PACKET_PARTICIPANTS_DATA_BUFFER_2019;
exports.PACKET_PARTICIPANTS_DATA_PARSED_2019 = PacketParticipantsData_2.PACKET_PARTICIPANTS_DATA_PARSED_2019;
const PacketSessionData_2 = require("./2019/PacketSessionData");
exports.PACKET_SESSION_DATA_BUFFER_2019 = PacketSessionData_2.PACKET_SESSION_DATA_BUFFER_2019;
exports.PACKET_SESSION_DATA_PARSED_2019 = PacketSessionData_2.PACKET_SESSION_DATA_PARSED_2019;
const PacketCarSetupData_3 = require("./2020/PacketCarSetupData");
exports.PACKET_CAR_SETUP_DATA_BUFFER_2020 = PacketCarSetupData_3.PACKET_CAR_SETUP_DATA_BUFFER_2020;
exports.PACKET_CAR_SETUP_DATA_PARSED_2020 = PacketCarSetupData_3.PACKET_CAR_SETUP_DATA_PARSED_2020;
const PacketCarStatusData_3 = require("./2020/PacketCarStatusData");
exports.PACKET_CAR_STATUS_DATA_BUFFER_2020 = PacketCarStatusData_3.PACKET_CAR_STATUS_DATA_BUFFER_2020;
exports.PACKET_CAR_STATUS_DATA_PARSED_2020 = PacketCarStatusData_3.PACKET_CAR_STATUS_DATA_PARSED_2020;
const PacketCarTelemetryData_3 = require("./2020/PacketCarTelemetryData");
exports.PACKET_CAR_TELEMETRY_DATA_BUFFER_2020 = PacketCarTelemetryData_3.PACKET_CAR_TELEMETRY_DATA_BUFFER_2020;
exports.PACKET_CAR_TELEMETRY_DATA_PARSED_2020 = PacketCarTelemetryData_3.PACKET_CAR_TELEMETRY_DATA_PARSED_2020;
const PacketEventData_3 = require("./2020/PacketEventData");
exports.PACKET_EVENT_DATA_BUFFER_2020 = PacketEventData_3.PACKET_EVENT_DATA_BUFFER_2020;
exports.PACKET_EVENT_DATA_PARSED_2020 = PacketEventData_3.PACKET_EVENT_DATA_PARSED_2020;
const PacketFinalClassificationData_1 = require("./2020/PacketFinalClassificationData");
exports.PACKET_FINAL_CLASSIFICATION_DATA_BUFFER_2020 = PacketFinalClassificationData_1.PACKET_FINAL_CLASSIFICATION_DATA_BUFFER_2020;
exports.PACKET_FINAL_CLASSIFICATION_DATA_PARSED_2020 = PacketFinalClassificationData_1.PACKET_FINAL_CLASSIFICATION_DATA_PARSED_2020;
const PacketHeader_3 = require("./2020/PacketHeader");
exports.PACKET_HEADER_BUFFER_2020 = PacketHeader_3.PACKET_HEADER_BUFFER_2020;
exports.PACKET_HEADER_PARSED_2020 = PacketHeader_3.PACKET_HEADER_PARSED_2020;
const PacketLapData_3 = require("./2020/PacketLapData");
exports.PACKET_LAP_DATA_BUFFER_2020 = PacketLapData_3.PACKET_LAP_DATA_BUFFER_2020;
exports.PACKET_LAP_DATA_PARSED_2020 = PacketLapData_3.PACKET_LAP_DATA_PARSED_2020;
const PacketLobbyInfoData_1 = require("./2020/PacketLobbyInfoData");
exports.PACKET_LOBBY_INFO_DATA_BUFFER_2020 = PacketLobbyInfoData_1.PACKET_LOBBY_INFO_DATA_BUFFER_2020;
exports.PACKET_LOBBY_INFO_DATA_PARSED_2020 = PacketLobbyInfoData_1.PACKET_LOBBY_INFO_DATA_PARSED_2020;
const PacketMotionData_3 = require("./2020/PacketMotionData");
exports.PACKET_MOTION_DATA_BUFFER_2020 = PacketMotionData_3.PACKET_MOTION_DATA_BUFFER_2020;
exports.PACKET_MOTION_DATA_PARSED_2020 = PacketMotionData_3.PACKET_MOTION_DATA_PARSED_2020;
const PacketParticipantsData_3 = require("./2020/PacketParticipantsData");
exports.PACKET_PARTICIPANTS_DATA_BUFFER_2020 = PacketParticipantsData_3.PACKET_PARTICIPANTS_DATA_BUFFER_2020;
exports.PACKET_PARTICIPANTS_DATA_PARSED_2020 = PacketParticipantsData_3.PACKET_PARTICIPANTS_DATA_PARSED_2020;
const PacketSessionData_3 = require("./2020/PacketSessionData");
exports.PACKET_SESSION_DATA_BUFFER_2020 = PacketSessionData_3.PACKET_SESSION_DATA_BUFFER_2020;
exports.PACKET_SESSION_DATA_PARSED_2020 = PacketSessionData_3.PACKET_SESSION_DATA_PARSED_2020;
const PacketCarSetupData_4 = require("./2021/PacketCarSetupData");
exports.PACKET_CAR_SETUP_DATA_BUFFER_2021 = PacketCarSetupData_4.PACKET_CAR_SETUP_DATA_BUFFER_2021;
exports.PACKET_CAR_SETUP_DATA_PARSED_2021 = PacketCarSetupData_4.PACKET_CAR_SETUP_DATA_PARSED_2021;
const PacketCarStatusData_4 = require("./2021/PacketCarStatusData");
exports.PACKET_CAR_STATUS_DATA_BUFFER_2021 = PacketCarStatusData_4.PACKET_CAR_STATUS_DATA_BUFFER_2021;
exports.PACKET_CAR_STATUS_DATA_PARSED_2021 = PacketCarStatusData_4.PACKET_CAR_STATUS_DATA_PARSED_2021;
const PacketCarTelemetryData_4 = require("./2021/PacketCarTelemetryData");
exports.PACKET_CAR_TELEMETRY_DATA_BUFFER_2021 = PacketCarTelemetryData_4.PACKET_CAR_TELEMETRY_DATA_BUFFER_2021;
exports.PACKET_CAR_TELEMETRY_DATA_PARSED_2021 = PacketCarTelemetryData_4.PACKET_CAR_TELEMETRY_DATA_PARSED_2021;
const PacketEventData_4 = require("./2021/PacketEventData");
exports.PACKET_EVENT_DATA_BUFFER_2021 = PacketEventData_4.PACKET_EVENT_DATA_BUFFER_2021;
exports.PACKET_EVENT_DATA_PARSED_2021 = PacketEventData_4.PACKET_EVENT_DATA_PARSED_2021;
const PacketHeader_4 = require("./2021/PacketHeader");
exports.PACKET_HEADER_BUFFER_2021 = PacketHeader_4.PACKET_HEADER_BUFFER_2021;
exports.PACKET_HEADER_PARSED_2021 = PacketHeader_4.PACKET_HEADER_PARSED_2021;
const PacketLapData_4 = require("./2021/PacketLapData");
exports.PACKET_LAP_DATA_BUFFER_2021 = PacketLapData_4.PACKET_LAP_DATA_BUFFER_2021;
exports.PACKET_LAP_DATA_PARSED_2021 = PacketLapData_4.PACKET_LAP_DATA_PARSED_2021;
const PacketMotionData_4 = require("./2021/PacketMotionData");
exports.PACKET_MOTION_DATA_BUFFER_2021 = PacketMotionData_4.PACKET_MOTION_DATA_BUFFER_2021;
exports.PACKET_MOTION_DATA_PARSED_2021 = PacketMotionData_4.PACKET_MOTION_DATA_PARSED_2021;
const PacketParticipantsData_4 = require("./2021/PacketParticipantsData");
exports.PACKET_PARTICIPANTS_DATA_BUFFER_2021 = PacketParticipantsData_4.PACKET_PARTICIPANTS_DATA_BUFFER_2021;
exports.PACKET_PARTICIPANTS_DATA_PARSED_2021 = PacketParticipantsData_4.PACKET_PARTICIPANTS_DATA_PARSED_2021;
const PacketSessionData_4 = require("./2021/PacketSessionData");
exports.PACKET_SESSION_DATA_BUFFER_2021 = PacketSessionData_4.PACKET_SESSION_DATA_BUFFER_2021;
exports.PACKET_SESSION_DATA_PARSED_2021 = PacketSessionData_4.PACKET_SESSION_DATA_PARSED_2021;
